### DISCUSSION TITLE
[Feature request] Differenciate Disable from Hide for queries

### DISCUSSION THREAD

**Discussion by luipir** [grafana/grafana#38147] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
IMHO Hide button for queries is misleading... it does not hide the query and related management of visualization, but disable query at all (this happen also in Transform tag->organize fields) => data are not restrieved or are not propagated to this.props.data.

**Why is this needed**:
Panels can display data depending on specific or ancillary queries that are only instruct the way to display

e.g. I've a table with color ranges mapping to have a dinamic color legend... color name column shouldn't be visible but used to create correct color steps.

**Comment by hugohaggmark** [comment]
Hi @luipir and thank you for adding this. I'm not sure I follow. Would the `visible` field override for a table field/column solve your issue?

**Comment by luipir** [comment]
> Hi @luipir and thank you for adding this. I'm not sure I follow. Would the `visible` field override for a table field/column solve your issue?

Hi @hugohaggmark, thanks for the hint, is it override in the option panel? if so, fields and override panels are available only for "new panels" wrote from scratch with new code.

BTW it's not only a problem of table visualization more than having ancillary queries that do not have to be shown at all but that it's result is used to setup visualization (e.g. variable value/color mapping)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
